flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -523,7 +523,7 @@ public void start() {
 								String.format("Failed to rescale the job %s.", jobGraph.getJobID()),
 								throwable)));
 				}
-			});
+			}, getMainThreadExecutor());
```

Comment:
This could happen as a follow up: I think there should be no need to add the main thread executor here. Instead `rescalingFuture` should always be completed from within the main thread.